consoles/sshVirtsh.pm

```diff
@@ -352,9 +353,16 @@ sub _copy_image_vmware ($self, $name, $backingfile, $file_basename, $vmware_open
 }
 
 sub _copy_image_else ($self, $file, $file_basename, $basedir) {
-    $self->run_cmd(sprintf("rsync -av '$file' '$basedir/%s'", $file_basename)) && die 'rsync failed';
+    if (-e $file_basename && defined which 'rsync') {    # utilize asset possibly cached by openQA worker
+        my %c = $self->get_ssh_credentials;
+        bmwqemu::diag "Syncing '$file' directly from worker host to $c{hostname}";
+        CORE::GLOBAL::system "RSYNC_PASSWORD='$c{password}' rsync -av '$file_basename' '$c{username}\@$c{hostname}:$basedir/$file_basename'";
+    }
+    else {
+        $self->run_cmd("rsync -av '$file' '$basedir/$file_basename'") && die 'rsync failed';
+    }
     if ($file_basename =~ /(.*)\.xz$/) {
-        $self->run_cmd(sprintf("nice ionice unxz -f -k '$basedir/%s'", $file_basename)) unless -e "$basedir$1";
+        $self->run_cmd("nice ionice unxz -f -k '$basedir/$file_basename'") unless -e "$basedir$1";
```

Comment:
The `-e "$basedir$1"` condition is checked on the wrong machine.

Reply:
Or do we want the condition in the first place? The compressed file may not match the old uncompressed file.

Reply:
That condition looks very wrong, indeed. I have actually just removed the `sprintf` here for consistency with other code but supposedly it is worthwhile to fix this while we're at it.

> Or do we want the condition in the first place?

Good question. I guess at this point the condition is likely always false so we can assume everything would be working without it. Supposedly I better just remove it completely.